### PR TITLE
Refactor CLI hint code and types, more CLI hint arguments

### PR DESCRIPTION
**Pull request by vinceatbluelabs**
There are three major changes in this PR:

* Move all load/unload code to use well-typed, pre-validated hints rather than raw input when we have a fully specified RecordsFormat.
* Introduce a TypedDict partially specified version of hints to use as input to and output from our hint sniffing code.
* Use the latter to flush out the remaining command line options in mvrec related to hints, and enhance our JSON schema to argparse translator to handle another case.

Some background:

Up until now, the only command line parameters for hints that we've accepted are those in the 'BootstrappingHints' type, which is typed as a TypedDict with a limited selection of keys.

We have an 'initial_hints' argument to various factory methods that accept this type.  Having these separate from an optional 'records_format' argument allowed us to be told whether to do hint inference or not and still be able to override hint inference when we need to.

Because we didn't have Literal[] types for the relevant hints yet, these arguments were hand-crafted, so I didn't build all of them out.  The ones I hand-crafted in were motivated by things the hint sniffing code didn't address and which commonly caused loading errors (generally empirically as folks ran into issues).

Later on I provided a 'variant' CLI argument that if provided would combine with the hints provided to fill in the 'records_format' argument to the factory methods.  This was a little limited, as we still only hand selected arguments available, but it was nice for dealing with, say, an uncompresssed version of a common variant, or a slight delta on a commonly used and abused variant like 'csv'.

Now that we have the ability to easily generate the additional hints, and have types for everything, I'd like to go ahead and fully extend out the hints supported by the command line.

I made this a TypedDict with all of the hint keys included to help people who use type checkers, but we do have to be careful, as this is an external interface - just because we declared a type doesn't mean people will send us well-typed things!  I'm casting those back down to our untyped dictionary type once we receive it so we don't make any bad assumptions handling that
data.

Before:

```sh
(mylibs-3.8.2) ]0;graybookprobroz@graybookpro:~$ mvrec file2table
usage: mvrec file2table [-h] [--source.variant SOURCE.VARIANT]
                        [--source.field_delimiter SOURCE.FIELD-DELIMITER] [--source.no_compression]
                        [--source.compression {BZIP,GZIP,LZO}] [--source.no_escape] [--source.escape {\}]
                        [--source.no_quoting] [--source.quoting {all,minimal,nonnumeric}]
                        [--source.encoding {UTF8,UTF16,UTF16LE,UTF16BE,LATIN1,CP1252}]
                        [--target.existing_table {delete_and_overwrite,truncate_and_overwrite,drop_and_recreate,append}]
                        [--target.drop_and_recreate_on_load_error]
                        source.filename target.db_name target.schema_name target.table_name
mvrec file2table: error: the following arguments are required: source.filename, target.db_name, target.schema_name, target.table_name
(mylibs-3.8.2) ]0;graybookprobroz@graybookpro:~$ 
```

After:

```sh
(records-mover-3.8.2) ]0;graybookprobroz@graybookpro:~/src/records-mover$ mvrec file2table
usage: mvrec file2table [-h] [--source.variant SOURCE.VARIANT]
                        [--source.datetimeformattz {YYYY-MM-DD HH:MI:SSOF,YYYY-MM-DD HH:MI:SS,YYYY-MM-DD HH24:MI:SSOF,MM/DD/YY HH24:MI}]
                        [--source.datetimeformat {YYYY-MM-DD HH24:MI:SS,YYYY-MM-DD HH:MI:SS,YYYY-MM-DD HH12:MI AM,MM/DD/YY HH24:MI}]
                        [--source.no_compression] [--source.compression {GZIP,BZIP,LZO}]
                        [--source.no_quoting] [--source.quoting {all,minimal,nonnumeric}]
                        [--source.no_escape] [--source.escape {\}]
                        [--source.encoding {UTF8,UTF16,UTF16LE,UTF16BE,UTF16BOM,UTF8BOM,LATIN1,CP1252}]
                        [--source.dateformat {YYYY-MM-DD,MM-DD-YYYY,DD-MM-YYYY,MM/DD/YY}]
                        [--source.timeonlyformat {HH12:MI AM,HH24:MI:SS}] [--source.no_doublequote]
                        [--source.doublequote] [--source.no_header_row] [--source.header_row]
                        [--source.quotechar SOURCE.QUOTECHAR]
                        [--source.record_terminator SOURCE.RECORD-TERMINATOR]
                        [--source.field_delimiter SOURCE.FIELD-DELIMITER]
                        [--target.existing_table {delete_and_overwrite,truncate_and_overwrite,drop_and_recreate,append}]
                        [--target.drop_and_recreate_on_load_error]
                        source.filename target.db_name target.schema_name target.table_name
mvrec file2table: error: the following arguments are required: source.filename, target.db_name, target.schema_name, target.table_name
(records-mover-3.8.2) ]0;graybookprobroz@graybookpro:~/src/records-mover$ 
```

### CODE REVIEW

File: records_mover/records/job/hints.py [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
 from ...utils.json_schema import JsonParameter, JsonSchemaDocument
 from typing import Optional
 
 
+SUPPORTED_HINT_NAMES = [
+    'field-delimiter', 'compression', 'escape', 'quoting', 'encoding', 'header-row'
+]
+
+
+# TODO: This class isn't needed anymore if we can just provide Hint class
```
Comment (bluelabsbutler) [inline records_mover/records/job/hints.py:12]: Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker

File: records_mover/records/job/schema.py [hunk 2]
```diff
@@ -1,17 +1,25 @@
...
 
 def method_to_json_schema(method: Callable[..., Any]) -> JsonSchema:
     special_handling: Dict[str, List[JsonParameter]] = {
         'google_cloud_creds': [JsonParameter('gcp_creds_name', JsonSchemaDocument('string'))],
         'db_engine': [JsonParameter('db_name', JsonSchemaDocument('string'))],
         'records_format': ([JsonParameter('variant', JsonSchemaDocument('string'), optional=True)] +
-                           [hint.schema for hint in SUPPORTED_HINTS]),
-        'initial_hints': [hint.schema for hint in SUPPORTED_HINTS],
+                           HINT_PARAMETERS),
+        'initial_hints': HINT_PARAMETERS,  # TODO: Downselect to initial_hints?
```
Comment (bluelabsbutler) [inline records_mover/records/job/schema.py:22]: Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker

File: records_mover/db/bigquery/load_job_config_options.py [hunk 3]
```diff
@@ -106,8 +107,10 @@ def load_job_config(unhandled_hints: Set[str],
     config.schema_update_options = None
 
     if isinstance(load_plan.records_format, DelimitedRecordsFormat):
+        hints = validate_partial_hints(load_plan.records_format.hints,
+                                       fail_if_cant_handle_hint=load_plan.processing_instructions.fail_if_cant_handle_hint)
```
Comment (bluelabsbutler) [inline records_mover/db/bigquery/load_job_config_options.py:111]: E501 line too long (123 > 100 characters)

File: records_mover/db/postgres/copy_options/common.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 from records_mover.utils import quiet_remove
-from records_mover.records.delimited import cant_handle_hint, RecordsHints
+from records_mover.records.records_format import DelimitedRecordsFormat
```
Comment (bluelabsbutler) [inline records_mover/db/postgres/copy_options/common.py:2]: F401 'records_mover.records.records_format.DelimitedRecordsFormat' imported but unused

File: records_mover/db/postgres/copy_options/csv.py [hunk 5]
```diff
@@ -1,13 +1,14 @@
 from records_mover.utils import quiet_remove
-from records_mover.records.delimited import cant_handle_hint, RecordsHints
+from records_mover.records.records_format import DelimitedRecordsFormat
```
Comment (bluelabsbutler) [inline records_mover/db/postgres/copy_options/csv.py:2]: F401 'records_mover.records.records_format.DelimitedRecordsFormat' imported but unused

File: records_mover/db/postgres/copy_options/text.py [hunk 6]
```diff
@@ -1,13 +1,14 @@
 from records_mover.utils import quiet_remove
-from records_mover.records.delimited import cant_handle_hint, RecordsHints
+from records_mover.records.records_format import DelimitedRecordsFormat
```
Comment (bluelabsbutler) [inline records_mover/db/postgres/copy_options/text.py:2]: F401 'records_mover.records.records_format.DelimitedRecordsFormat' imported but unused

File: records_mover/records/delimited/__init__.py [hunk 7]
```diff
@@ -34,8 +35,9 @@
 hints.
 """
 
-from .types import BootstrappingRecordsHints, RecordsHints, MutableRecordsHints
+from .types import BootstrappingRecordsHints, RecordsHints, MutableRecordsHints, UntypedRecordsHints
 from .validated_records_hints import ValidatedRecordsHints
+from .hints import validate_partial_hints
```
Comment (bluelabsbutler) [inline records_mover/records/delimited/__init__.py:40]: F401 '.hints.validate_partial_hints' imported but unused

File: records_mover/records/delimited/hint.py [hunk 8]
```diff
@@ -1,29 +1,43 @@
 from typing_inspect import is_literal_type, get_args
 from abc import ABCMeta, abstractmethod
-from .types import HintName, RecordsHints
-from typing import TypeVar, Generic, Type, List
+from .types import HintName, RecordsHints, UntypedRecordsHints
```
Comment (bluelabsbutler) [inline records_mover/records/delimited/hint.py:3]: F401 '.types.RecordsHints' imported but unused

File: records_mover/records/delimited/types.py [hunk 9]
```diff
@@ -1,12 +1,14 @@
-from typing import Optional, Union, Mapping, Dict
-from typing_extensions import Literal, TypedDict
+from typing_inspect import get_args, typed_dict_keys
+from typing import Optional, Union, Mapping, Dict, List
```
Comment (bluelabsbutler) [inline records_mover/records/delimited/types.py:2]: F401 'typing.Optional' imported but unused

Comment (bluelabsbutler) [inline records_mover/records/delimited/types.py:2]: F401 'typing.Union' imported but unused

File: records_mover/records/sources/uninferred_fileobjs.py [hunk 10]
```diff
@@ -4,7 +4,9 @@
 from ..schema import RecordsSchema
 from ..processing_instructions import ProcessingInstructions
 from ..records_format import BaseRecordsFormat
-from .. import BootstrappingRecordsHints
+from records_mover.records.delimited import (
```
Comment (bluelabsbutler) [inline records_mover/records/sources/uninferred_fileobjs.py:7]: F401 'records_mover.records.delimited.BootstrappingRecordsHints' imported but unused

Comment (bluelabsbutler) [inline records_mover/records/sources/uninferred_fileobjs.py:7]: F401 'records_mover.records.delimited.RecordsHints' imported but unused

File: records_mover/records/sources/uninferred_fileobjs.py [hunk 11]
```diff
@@ -16,7 +18,7 @@ def __init__(self,
                  target_names_to_input_fileobjs: Mapping[str, IO[bytes]],
                  records_format: Optional[BaseRecordsFormat]=None,
                  records_schema: Optional[RecordsSchema]=None,
-                 initial_hints: Optional[BootstrappingRecordsHints]=None) -> None:
+                 initial_hints: Optional[UntypedRecordsHints]=None) -> None:
```
Comment (bluelabsbutler) [inline records_mover/records/sources/uninferred_fileobjs.py:21]: E252 missing whitespace around parameter equals

File: records_mover/records/delimited/hints.py [hunk 12]
```diff
@@ -19,34 +22,82 @@ class Hints:
...
+                                        'or which contain new-line characters.'))
+    record_terminator = StringHint('record-terminator',
+                                   default='\n',
+                                   description='String used to close out individual rows of data.')
+    field_delimiter = StringHint('field-delimiter',
+                                 default=',',
+                                 description='Character used between fields.')
+
+
+# TODO: Shoudl this be a method on DelimitedRecordsFormat as well?
```
Comment (bluelabsbutler) [inline records_mover/records/delimited/hints.py:94]: Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker

File: records_mover/records/delimited/types.py [hunk 13]
```diff
@@ -71,3 +58,44 @@
                    "timeonlyformat",
                    "datetimeformattz",
                    "datetimeformat"]
+
+HINT_NAMES: List[HintName] = list(get_args(HintName))  # type: ignore
+
+# TODO: Retire this
```
Comment (bluelabsbutler) [inline records_mover/records/delimited/types.py:64]: Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker

File: records_mover/records/delimited/types.py [hunk 14]
```diff
@@ -71,3 +58,44 @@
...
+                                          'header-row': HintHeaderRow,
+                                          'field-delimiter': HintFieldDelimiter,
+                                          'encoding': HintEncoding,
+                                          'escape': HintEscape,
+                                          'compression': HintCompression,
+                                          'record-terminator': HintRecordTerminator,
+                                      },
+                                      total=False)
+
+# TODO: Retire this
```
Comment (bluelabsbutler) [inline records_mover/records/delimited/types.py:77]: Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker

File: records_mover/records/delimited/types.py [hunk 15]
```diff
@@ -71,3 +58,44 @@
...
+                                    'record-terminator': HintRecordTerminator,
+                                    'quotechar': HintQuoteChar,
+                                    'doublequote': HintDoublequote,
+                                    'dateformat': HintDateFormat,
+                                    'timeonlyformat': HintTimeOnlyFormat,
+                                    'datetimeformattz': HintDateTimeFormatTz,
+                                    'datetimeformat': HintDateTimeFormat,
+                                },
+                                total=False)
+# TODO: Rename this throughout?
```
Comment (bluelabsbutler) [inline records_mover/records/delimited/types.py:97]: Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker

File: records_mover/records/delimited/utils.py [hunk 16]
```diff
@@ -1,15 +1,19 @@
...
 
 
 def complain_on_unhandled_hints(fail_if_dont_understand: bool,
                                 unhandled_hints: Iterable[str],
-                                hints: RecordsHints) -> None:
-    unhandled_bindings = [f"{k}={hints[k]}" for k in unhandled_hints]
+                                hints: Union[RecordsHints,
+                                             UntypedRecordsHints,
+                                             ValidatedRecordsHints]) -> None:
+    # TODO: this probably doesn't handle ValidatedRecordsHints yet
```
Comment (bluelabsbutler) [inline records_mover/records/delimited/utils.py:15]: Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker

File: records_mover/records/delimited/utils.py [hunk 17]
```diff
@@ -20,11 +24,17 @@ def complain_on_unhandled_hints(fail_if_dont_understand: bool,
             logger.warning(f"Did not understand these hints: {unhandled_bindings_str}")
 
 
-def cant_handle_hint(fail_if_cant_handle_hint: bool, hint_name: str, hints: RecordsHints) -> None:
+def cant_handle_hint(fail_if_cant_handle_hint: bool,
+                     hint_name: str,
+                     hints: Union[RecordsHints,
+                                  UntypedRecordsHints,
+                                  ValidatedRecordsHints]) -> None:
+    # TODO: this probably doesn't handle ValidatedRecordsHints yet
```
Comment (bluelabsbutler) [inline records_mover/records/delimited/utils.py:32]: Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker

File: records_mover/records/delimited/types.py [hunk 18]
```diff
@@ -1,12 +1,14 @@
-from typing import Optional, Union, Mapping, Dict
-from typing_extensions import Literal, TypedDict
+from typing_inspect import get_args, typed_dict_keys
```
Comment (bluelabsbutler) [inline records_mover/records/delimited/types.py:1]: F401 'typing_inspect.typed_dict_keys' imported but unused

File: records_mover/records/sources/data_url.py [hunk 19]
```diff
@@ -20,7 +20,7 @@ def __init__(self,
                  url_resolver: UrlResolver,
                  records_format: Optional[BaseRecordsFormat]=None,
                  records_schema: Optional[RecordsSchema]=None,
-                 initial_hints: Optional[BootstrappingRecordsHints]=None) -> None:
+                 initial_hints: Optional[RecordsHints]=None) -> None:
```
Comment (bluelabsbutler) [inline records_mover/records/sources/data_url.py:23]: E252 missing whitespace around parameter equals

File: records_mover/records/sources/factory.py [hunk 20]
```diff
@@ -91,7 +91,7 @@ def dataframes(self,
     def fileobjs(self,
                  target_names_to_input_fileobjs: Mapping[str, IO[bytes]],
                  records_format: Optional[BaseRecordsFormat]=None,
-                 initial_hints: Optional[BootstrappingRecordsHints]=None,
+                 initial_hints: Optional[RecordsHints]=None,
```
Comment (bluelabsbutler) [inline records_mover/records/sources/factory.py:94]: E252 missing whitespace around parameter equals

File: records_mover/records/sources/factory.py [hunk 21]
```diff
@@ -118,7 +118,7 @@ def fileobjs(self,
     def data_url(self,
                  input_url: str,
                  records_format: Optional[BaseRecordsFormat]=None,
-                 initial_hints: Optional[BootstrappingRecordsHints]=None,
+                 initial_hints: Optional[RecordsHints]=None,
```
Comment (bluelabsbutler) [inline records_mover/records/sources/factory.py:121]: E252 missing whitespace around parameter equals

File: records_mover/records/sources/factory.py [hunk 22]
```diff
@@ -179,7 +179,7 @@ def directory_from_url(self,
     def local_file(self,
                    filename: str,
                    records_format: Optional[BaseRecordsFormat]=None,
-                   initial_hints: Optional[BootstrappingRecordsHints]=None,
+                   initial_hints: Optional[RecordsHints]=None,
```
Comment (bluelabsbutler) [inline records_mover/records/sources/factory.py:182]: E252 missing whitespace around parameter equals

File: tests/unit/db/redshift/test_redshift_db_driver_unload.py [hunk 23]
```diff
@@ -131,25 +136,27 @@ def test_unload_christmas_tree_unsupported_options_with_fast_warns_2(self,
...
-                                   call("Ignoring hint record-terminator = '\\x02'"),
-                                   call("Ignoring hint datetimeformat = None"),
-                                   call("Ignoring hint dateformat = 'MM-DD-YYYY'")])
+            self.assertListEqual(mock_warning.mock_calls,
+                                 [call("Ignoring hint escape = '@'"),
+                                  call("Ignoring hint datetimeformattz = 'HH:MI:SSOF YYYY-MM-DD'"),
+                                  call("Ignoring hint record-terminator = '\\x02'"),
+                                  call("Ignoring hint doublequote = True"),
+                                  call("Ignoring hint compression = 'BZIP'"),
+                                  call("Ignoring hint datetimeformattz = 'YYYY-MM-DD HH24:MI:SSOF'"),
```
Comment (bluelabsbutler) [inline tests/unit/db/redshift/test_redshift_db_driver_unload.py:154]: E501 line too long (101 > 100 characters)

File: tests/unit/records/test_pandas_to_csv_options_christmas_tree.py [hunk 24]
```diff
@@ -79,19 +80,23 @@ def test_pandas_to_csv_options_christmas_tree_format_3(self):
...
-                                  [call("Ignoring hint datetimeformat = None"),
-                                   call("Ignoring hint datetimeformattz = 'HH:MI:SSOF YYYY-MM-DD'"),
-                                   call("Ignoring hint quoting = "
-                                        "'some_future_option_not_supported_now'")])
+            self.assertListEqual(mock_warning.mock_calls,
+                                 [call("Ignoring hint quoting = "
+                                       "'some_future_option_not_supported_now'"),
+                                  call("Ignoring hint escape = '@'"),
+                                  call("Ignoring hint datetimeformattz = 'HH:MI:SSOF YYYY-MM-DD'"),
+                                  call("Ignoring hint datetimeformattz = 'YYYY-MM-DD HH24:MI:SSOF'"),
```
Comment (bluelabsbutler) [inline tests/unit/records/test_pandas_to_csv_options_christmas_tree.py:88]: E501 line too long (101 > 100 characters)

File: records_mover/records/delimited/__init__.py [hunk 25]
```diff
@@ -34,8 +34,9 @@
 hints.
 """
 
-from .types import BootstrappingRecordsHints, RecordsHints, MutableRecordsHints
+from .types import PartialRecordsHints, MutableUntypedRecordsHints, UntypedRecordsHints
```
Comment (bluelabsbutler) [inline records_mover/records/delimited/__init__.py:37]: F401 '.types.PartialRecordsHints' imported but unused

Comment (bluelabsbutler) [inline records_mover/records/delimited/__init__.py:37]: F401 '.types.MutableUntypedRecordsHints' imported but unused

File: records_mover/records/delimited/hint.py [hunk 26]
```diff
@@ -1,29 +1,43 @@
 from typing_inspect import is_literal_type, get_args
 from abc import ABCMeta, abstractmethod
-from .types import HintName, RecordsHints
-from typing import TypeVar, Generic, Type, List
+from .types import HintName, PartialRecordsHints, UntypedRecordsHints
```
Comment (bluelabsbutler) [inline records_mover/records/delimited/hint.py:3]: F401 '.types.PartialRecordsHints' imported but unused

File: records_mover/records/records_format.py [hunk 27]
```diff
@@ -1,9 +1,9 @@
 import logging
 from .processing_instructions import ProcessingInstructions
-from . import RecordsHints
+from . import PartialRecordsHints, UntypedRecordsHints
 from .base_records_format import BaseRecordsFormat
 from typing import Mapping, Optional, TYPE_CHECKING
-from .delimited import MutableRecordsHints, ValidatedRecordsHints
+from .delimited import MutableUntypedRecordsHints, ValidatedRecordsHints
```
Comment (bluelabsbutler) [inline records_mover/records/records_format.py:6]: F401 '.delimited.MutableUntypedRecordsHints' imported but unused

File: records_mover/records/records_format.py [hunk 28]
```diff
@@ -26,14 +26,17 @@ def generate_filename(self, basename: str) -> str:
...
+    # as it may come from public interface consumers.
+    hints: UntypedRecordsHints
     """Stores the full set of hints describing the format, combining both
     the default hints for the variant and any hint overrides provided
     in the constructor"""
 
     def __init__(self,
                  variant: str='bluelabs',
-                 hints: RecordsHints={},
+                 hints: PartialRecordsHints={},
```
Comment (bluelabsbutler) [inline records_mover/records/records_format.py:39]: E252 missing whitespace around parameter equals

File: records_mover/records/records_format.py [hunk 29]
```diff
@@ -191,7 +195,7 @@ def __repr__(self) -> str:
 
 def RecordsFormat(format_type: 'RecordsFormatType' = 'delimited',
                   variant: str='bluelabs',
-                  hints: RecordsHints={},
+                  hints: PartialRecordsHints={},
```
Comment (bluelabsbutler) [inline records_mover/records/records_format.py:198]: E252 missing whitespace around parameter equals

File: records_mover/records/sources/directory.py [hunk 30]
```diff
@@ -17,7 +17,7 @@ def __init__(self,
                  directory: RecordsDirectory,
                  fail_if_dont_understand: bool,
                  url_resolver: UrlResolver,
-                 override_hints: RecordsHints={}) -> None:
+                 override_hints: PartialRecordsHints={}) -> None:
```
Comment (bluelabsbutler) [inline records_mover/records/sources/directory.py:20]: E252 missing whitespace around parameter equals

File: records_mover/records/sources/factory.py [hunk 31]
```diff
@@ -179,7 +179,7 @@ def directory_from_url(self,
     def local_file(self,
                    filename: str,
                    records_format: Optional[BaseRecordsFormat]=None,
-                   initial_hints: Optional[BootstrappingRecordsHints]=None,
+                   initial_hints: Optional[PartialRecordsHints]=None,
```
Comment (bluelabsbutler) [inline records_mover/records/sources/factory.py:182]: E252 missing whitespace around parameter equals

File: records_mover/db/bigquery/load_job_config_options.py [hunk 32]
```diff
@@ -121,7 +124,8 @@ def load_job_config(unhandled_hints: Set[str],
 
 
 def add_load_job_csv_config(unhandled_hints: Set[str],
-                            hints: RecordsHints,
+                            # TODO: Use validated hints
```
Comment (bluelabsbutler) [inline records_mover/db/bigquery/load_job_config_options.py:127]: Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker

File: records_mover/records/records_format.py [hunk 33]
```diff
@@ -60,20 +63,21 @@ def __eq__(self, other: object) -> bool:
...
         return DelimitedRecordsFormat(variant=self.variant,
-                                      hints=input_hints)
+                                      hints=input_hints)  # type: ignore
 
     def alter_variant(self, variant: str) -> 'DelimitedRecordsFormat':
         return DelimitedRecordsFormat(variant=variant,
-                                      hints=self.hints)
+                                      hints=self.hints)  # type: ignore
 
+    # TODO: Can this be made to return PartialRecordsHints?
```
Comment (bluelabsbutler) [inline records_mover/records/records_format.py:77]: Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker

File: tests/unit/url/test_resolver_misconfigured.py [hunk 34]
```diff
@@ -0,0 +1,40 @@
+from records_mover.url.resolver import file_url_ctors, directory_url_ctors, UrlResolver
+from records_mover.url.base import BaseFileUrl, BaseDirectoryUrl
+from mock import Mock
```
Comment (bluelabsbutler) [inline tests/unit/url/test_resolver_misconfigured.py:3]: F401 'mock.Mock' imported but unused

File: metrics/coverage_high_water_mark [hunk 35]
```diff
@@ -1 +1 @@
-94.0200
\ No newline at end of file
+93.6600
```
Comment (vinceatbluelabs) [inline metrics/coverage_high_water_mark:1]: If you zoom down to the tests, I had to disable a few tests - we had code and tests supporting hint values for date/time values which don't actually exist in the spec.  Now that we're doing validation using the correct list of supported hint values, those tests can't be run.

Adding those hint values makes sense, for sure, though.  Adding those hint values and making sure all the different drivers support them correctly is likely to spiral just a bit from this already 56 file PR, so I'm proposing to comment out those tests and leave those hints as unsupported.  I have a high priority follow-up backlog task to handle these.  It's not normal at least in internal use to be passing in date/time hints, so I'm not terribly concerned about breaking any existing out-of-spec code with this.

Commented out tests:
* https://github.com/bluelabsio/records-mover/pull/66/files#diff-b9d3d8d84af11c791565c585049c18be
* https://github.com/bluelabsio/records-mover/pull/66/files#diff-4abbf34edf14477a63b4fe26b409b96e

Backlog task: https://app.asana.com/0/1128138765527694/1178155809546696

File: records_mover/records/delimited/hint.py [hunk 36]
```diff
@@ -1,29 +1,43 @@
 from typing_inspect import is_literal_type, get_args
```
Comment (vinceatbluelabs) [inline records_mover/records/delimited/hint.py:1]: Let's teach the Hint type to have descriptions and be able to generate JSON schema information!

File: records_mover/records/job/hints.py [hunk 37]
```diff
@@ -1,66 +0,0 @@
-"""Defines hints supported by the job config parser."""
```
Comment (vinceatbluelabs) [inline records_mover/records/job/hints.py:1]: This is the duplicative descriptions of hints that are no longer needed!

File: metrics/coverage_high_water_mark [hunk 38]
```diff
@@ -1 +1 @@
-94.0400
\ No newline at end of file
+93.6800
```
Comment (vinceatbluelabs) [inline metrics/coverage_high_water_mark:1]: If you zoom down to the tests, I had to disable a few tests - we had code and tests supporting hint values for date/time values which don't actually exist in the spec. Now that we're doing validation using the correct list of supported hint values, those tests can't be run.

Adding those hint values makes sense, for sure, though. Adding those hint values and making sure all the different drivers support them correctly is likely to spiral just a bit from this already 56 file PR, so I'm proposing to comment out those tests and leave those hints as unsupported. I have a high priority follow-up backlog task to handle these. It's not normal at least in internal use to be passing in date/time hints, so I'm not terribly concerned about breaking any existing out-of-spec code with this.

Commented out tests:

https://github.com/bluelabsio/records-mover/pull/66/files#diff-b9d3d8d84af11c791565c585049c18be
https://github.com/bluelabsio/records-mover/pull/66/files#diff-4abbf34edf14477a63b4fe26b409b96e
Backlog task: https://app.asana.com/0/1128138765527694/1178155809546696

**Pull request merged**